### PR TITLE
adding Group layer support for layerByName

### PR DESCRIPTION
**Pull request by kurtome**
Previously `map.layerByName` only searched the topmost layer, but it should additionally search through `Group` layers which have child `layers`.

For example:

```dart
map.layerByName('nested_layer');
```

Should return a `TileLayer` with TMX XML similar too this:
```xml
 <group id="14" name="group_layer">
  <layer id="2" name="nested_layer" width="46" height="34">
  <layer id="3" name="other_nested_layer" width="46" height="34">
 </group>
```

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -1,3 +1,6 @@
+## 0.8.5
```
Comment (spydon) [inline CHANGELOG.md:1]: This should say [Next] (We really should introduce melos on this repo too...)

**Review by spydon — APPROVED** [review]

File: CHANGELOG.md (hunk 1 above)
Comment (kurtome) [reply CHANGELOG.md]: fixed

**Comment by spydon** [comment]
Can you fix the little merge conflict here?
I'll set up melos for this repository when I get home if I have time, so that we don't get these changelog merge conflicts.

**Comment by kurtome** [comment]
@spydon fixed the merge conflict

**Pull request merged**